### PR TITLE
Change save directory into `outdir`

### PR DESCRIPTION
**Pull request by DavdGao**
as the title says

### CODE REVIEW

File: federatedscope/core/worker/client.py [hunk 1]
```diff
@@ -445,15 +445,12 @@ def callback_funcs_for_finish(self, message: Message):
...
         # Save final prediction result
         if self._cfg.data.type == 'cikmcup':
             # Evaluate
             self.trainer.evaluate(target_data_split_name='test')
-            # Save results
-            self.trainer.save_prediction(self._cfg.eval.prediction_path, self.ID, self._cfg.model.task)
-            logger.info(f"Client #{self.ID} finished saving prediction results in {os.path.abspath(self._cfg.eval.prediction_path)}")
-            logger.info(f"Please remember to save the prediction results elsewhere to avoid being covered by the new prediction results.")
+            self.trainer.save_prediction(self._cfg.outdir, self.ID, self._cfg.model.task)
+            logger.info(f"Client #{self.ID} finished saving prediction results in {os.path.abspath(self._cfg.outdir)}/predictions.csv")
```
Comment (joneswong) [inline federatedscope/core/worker/client.py:453]: we'd better use `os.path.join` to concatenate them

**Review by joneswong — COMMENTED** [review]
Please update the guidance accordingly

File: federatedscope/core/worker/client.py (hunk 1 above)
Comment (DavdGao) [reply federatedscope/core/worker/client.py]: > we'd better use `os.path.join` to concatenate them

Modified accordingly

**Comment by DavdGao** [comment]
> Please update the guidance accordingly

Done

**Review by joneswong — APPROVED** [review]
LGTM

**Review by Osier-Yi — APPROVED** [review]
LGTM

**Pull request merged**